internal/api/handlers/books/get_book_by_id.go

```diff
@@ -0,0 +1,65 @@
+package books
+
+import (
+	"math"
+	"net/http"
+	"strings"
+	"test-project/internal/api"
+	"test-project/internal/types"
+	"test-project/internal/util"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/labstack/echo/v4"
+)
+
+func GetBookByIDRoute(s *api.Server) *echo.Route {
+	handler := newHandler(s.Books)
+	return s.Router.APIV1Book.GET("/:book_id", handler.GetBookByID())
+}
+
+func (h *Handler) GetBookByID() echo.HandlerFunc {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		bookID := c.Param("book_id")
+		if bookID == "" {
+			return c.JSON(http.StatusBadRequest, map[string]string{
+				"error": "book ID is required",
+			})
+		}
+
+		res, err := h.service.GetBookByID(ctx, bookID)
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
```

Comment:
du kannst mit errors.Is auf error typen prüfen, oder du lieferst vom service gleich den httperror zurück